libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -3917,6 +3917,30 @@ void GCS_MAVLINK::send_mount_status() const
     mount->send_mount_status(chan);
 }
 
+void GCS_MAVLINK::send_set_position_target_global_int(uint8_t target_system, uint8_t target_component, const Location& loc)
+{
+
+    uint16_t type_mask = POSITION_TARGET_TYPEMASK_VX_IGNORE | POSITION_TARGET_TYPEMASK_VY_IGNORE | POSITION_TARGET_TYPEMASK_VZ_IGNORE | \
+                         POSITION_TARGET_TYPEMASK_AX_IGNORE | POSITION_TARGET_TYPEMASK_AY_IGNORE | POSITION_TARGET_TYPEMASK_AZ_IGNORE | \
+                         POSITION_TARGET_TYPEMASK_YAW_IGNORE | POSITION_TARGET_TYPEMASK_YAW_RATE_IGNORE;
+
+    uint8_t mav_frame = loc.relative_alt ? MAV_FRAME_GLOBAL_RELATIVE_ALT_INT : MAV_FRAME_GLOBAL_INT;
```

Comment:
relative to what?  We need the inverse function for `mavlink_coordinate_frame_to_location_alt_frame`.

`loc.alt` should be put into the desired frame with `get_alt_cm(...)`

Reply:
ok txs.  I've changed it to always send relative to the home altitude.  It's a bit of a non-issue for now because this is only implemented by Rover and the mavros side throws away the altitude (because the path planner only works in 2D)..